File: src/geom/Point2d.ts
/**
 * A Point2D object represents a point in a 2D space.
 * @class Point2D
 * @since 0.1.0
 */
export class Point2D {

	//--------------------------------------------------------------------------
	// Properties
	//--------------------------------------------------------------------------

	/**
	 * The value of the point on the X axis.
	 * @property x
	 * @since 0.1.0
	 */
	public x: number = 0

	/**
	 * The value of the point on the Y axis.
	 * @property y
	 * @since 0.1.0
	 */
	public y: number = 0

	//--------------------------------------------------------------------------
	// Methods
	//--------------------------------------------------------------------------

	/**
	 * Initializes the point.
	 * @constructor
	 * @since 0.1.0
	 */
	constructor(x: number, y: number) {
		this.x = x
		this.y = y
	}

	/**
	 * @method round
	 * @since 0.1.0
	 */
	public round() {
		this.x = Math.round(this.x)
		this.y = Math.round(this.y)
		return this
	}

	/**
	 * @method floor
	 * @since 0.1.0
	 */
	public floor() {
		this.x = Math.floor(this.x)
		this.y = Math.floor(this.y)
		return this
	}

	/**
	 * @method ceil
	 * @since 0.1.0
	 */
	public ceil() {
		this.x = Math.ceil(this.x)
		this.y = Math.ceil(this.y)
		return this
	}
}

File: platforms/ios/Dezel/Dezel/View/Geom/Point2D.swift

/**
 * @class Point2D
 * @since 0.2.0
 */
public class Point2D {

	//--------------------------------------------------------------------------
	// MARK: Properties
	//--------------------------------------------------------------------------

	/**
	 * @property x
	 * @since 0.2.0
	 */
	public var x: Double = 0

	/**
	 * @property y
	 * @since 0.2.0
	 */
	public var y: Double = 0

	//--------------------------------------------------------------------------
	// MARK: Methods
	//--------------------------------------------------------------------------

	/**
	 * @constructor
	 * @since 0.2.0
	 */
	public init(x: Double, y: Double) {
		self.x = x
		self.y = y
	}

	/**
	 * @method round
	 * @since 0.2.0
	 */
	public func round() {
		self.x.round()
		self.y.round()
	}

	/**
	 * @method floor
	 * @since 0.1.0
	 */
	public func floor() {
		self.x.round(.down)
		self.y.round(.down)
	}

	/**
	 * @method ceil
	 * @since 0.1.0
	 */
	public func ceil() {
		self.x.round(.up)
		self.y.round(.up)
	}
}
